feat: add auto-triage via Backstage

simaarfania opened the pull request:
Related to Hackathon project which is built off of https://github.com/grafana/deployment_tools/issues/239391.


- **Triggers**: When issues are opened with "Affected Resource(s)" 
- **Queries**: Backstage API to find resource ownership -- *needs to be tested**
- **Assigns**: Issues to team GitHub projects automatically **needs to be tested**
- **Labels**: Adds `team/[name]` labels  
- **Notifies**: Mentions owning teams in comments

simaarfania committed:
feat: add auto-triage via Backstage

spinillos committed:
Fix security issues

spinillos committed:
Update hashes

spinillos committed:
Get secrets from vault

spinillos committed:
Login with service account

spinillos committed:
Remove login to gar

Duologic committed:
fix(backstage-lookup): traverse relations to look for GitHub projects

spinillos commented on scripts/backstage-lookup/main.go:
```diff
@@ -0,0 +1,244 @@
+
+// Constants for API and configuration
+const (
+	defaultURL     = "https://enghub.grafana-ops.net"
```
Should we get the value from the vault? I don't know if we should expose this...

simaarfania replied:
I think I've done that now! Let me know if that's the best way to do it.

simaarfania committed:
fix: source backstage URL from secrets vault

spinillos committed:
Pass access token to the script as env

Duologic committed:
feat: add functions to add issues to GitHub projects

Duologic committed:
refactor: use go-backstage lib instead of homegrown

Duologic committed:
docs: fix variable name

simaarfania committed:
feat: add verbose error handling

Duologic committed:
feat: add dry-run flag

Duologic committed:
fix: allow overriding the groupRef

Duologic committed:
feat(backstage-lookup): authenticate to EngHub through GCP IAP

Replace BACKSTAGE_TOKEN with IAP-based authentication using
google.golang.org/api/idtoken. When IAP_AUDIENCE is set, the tool
obtains OIDC ID tokens via Application Default Credentials. When
unset, requests are unauthenticated (for local dev via port-forward).

Update the autoassign workflow to use ADC from the GCP auth step
instead of an explicit access token, and fix the Go version to match
go.mod.

Duologic committed:
ci(autoassign): add smoke-test job on PRs to validate IAP auth chain

Trigger the workflow on pull_request when backstage-lookup code or the
workflow itself changes. The smoke-test job builds the tool, authenticates
via WIF, and runs a dry-run lookup against a known resource to validate
the full chain: build, IAP auth, and Backstage connectivity.

Also adds id-token:write permission needed for Workload Identity
Federation.

Duologic committed:
fix(backstage-lookup): exit non-zero when all resource lookups fail

The tool previously swallowed lookup errors and exited 0 even when the
Backstage API returned errors (e.g. 404). Now do() returns an error
when all resource lookups fail, causing the process to exit non-zero.
Partial success (some lookups fail, others succeed) still exits 0.

Duologic committed:
fix(autoassign): move permissions to job level to fix zizmor excessive-permissions

Duologic commented:
The logic from this PR has been swallowed by https://github.com/grafana/enghub-github-issue-assigner
